polyply/src/apply_termini.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2024 University of Groningen
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from vermouth.log_helpers import StyleAdapter, get_logger
+LOGGER = StyleAdapter(get_logger(__name__))
+
+def apply_terminal_mod(meta_molecule, term, res):
+    """
+
+    Apply a terminal modification.
+    Note this assumes that the modification is additive, ie. no atoms are
+    removed
+
+    Parameters
+    ----------
+    meta_molecule: :class:`polyply.src.meta_molecule.MetaMolecule`
+    term: str
+        name modification in forcefield to apply
+    res: int
+        resid of where to apply the modification
+
+    Returns
+    ----------
+    meta_molecule
+    """
+
+    molecule = meta_molecule.molecule
+
+    mod_atoms = {}
+    for i in molecule.force_field.modifications[term].atoms:
+        if 'replace' in i:
+            mod_atoms[i['atomname']] = i['replace']
+        else:
+            mod_atoms[i['atomname']] = {}
+
+
+    anum_dict = {}
+    for atomid, node in enumerate(meta_molecule.molecule.nodes):
+        if meta_molecule.molecule.nodes[node]['resid'] == res:
```

Comment:
in meta-molecules all the resid are unique so there is no need to loop over all of them. In general `node_id = resid -1` but you can also get the node to resid dict (`nx.get_node_attributes` once and then simply pick the node.